### PR TITLE
Subtract tx fee from uxto

### PR DESCRIPTION
**Pull request by frogchocolatebarpie**

### CODE REVIEW

**Comment by visopsys** [comment]
The Tx fee calculation should subtract value from Tx size besides the the multi asset fee. This PR also needs tests for calculation.

File: x/sisu/cardano/tx_builder.go [hunk 1]
```diff
@@ -59,6 +62,29 @@ func BuildTx(node cardano.Node, network cardano.Network, sender, receiver cardan
 	amount.Coin = minUTXO
 	log.Debug("amount.Coin = ", amount.Coin)
 
+	// Subtract transaction fee from multi-asset amount
+	log.Debugf("token price for token %s = %d ", token.Id, token.Price)
+	txFeeInToken := helper.GetCardanoTxFeeInToken(big.NewInt(adaPrice), big.NewInt(token.Price), new(big.Int).SetUint64(uint64(minUTXO)))
```
Comment (visopsys) [inline x/sisu/cardano/tx_builder.go:67]: tokenPrice has 10^9 unit while ADA has 10^6. You need to convert between the 2.

Comment (frogchocolatebarpie) [reply x/sisu/cardano/tx_builder.go]: I don't understand your comment. Why unit of token price is matter in this calculation?

Comment (visopsys) [reply x/sisu/cardano/tx_builder.go]: You are converting from a 10^9 base (token.Price) to 10^6 base (token price). You need to have a conversion between the 2.

**Pull request closed without merging**